src/main/java/net/dv8tion/jda/api/entities/MessageSticker.java

```diff
@@ -0,0 +1,206 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.dv8tion.jda.api.entities;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Set;
+
+/**
+ * An object representing a sticker in a Discord message.
+ *
+ * @see Message#getStickers()
+ */
+public class MessageSticker implements ISnowflake
+{
+    private final long id;
+    private final String name;
+    private final String description;
+    private final long packId;
+    private final String asset;
+    private final String previewAsset;
+    private final StickerFormat formatType;
+    private final Set<String> tags;
+
+    /** Template for {@link #getAssetUrl()} */
+    public static final String ASSET_URL = "https://cdn.discordapp.com/stickers/%s/%s.%s";
+
+    public MessageSticker(final long id, final String name, final String description, final long packId, final String asset, final String previewAsset, final StickerFormat formatType, final Set<String> tags)
+    {
+        this.id = id;
+        this.name = name;
+        this.description = description;
+        this.packId = packId;
+        this.asset = asset;
+        this.previewAsset = previewAsset;
+        this.formatType = formatType;
+        this.tags = tags;
+    }
+
+    @Override
+    public long getIdLong()
+    {
+        return id;
+    }
+
+    /**
+     * The name of the sticker.
+     *
+     * @return the name of the sticker
+     */
+    @Nonnull
+    public String getName()
+    {
+        return name;
+    }
+
+    /**
+     * The description of the sticker.
+     *
+     * @return the description of the sticker
+     */
+    @Nonnull
+    public String getDescription()
+    {
+        return description;
+    }
+
+    /**
+     * The ID of the pack the sticker is from.
+     *
+     * @return the ID of the pack the sticker is from
+     */
+    @Nonnull
+    public String getPackId()
+    {
+        return Long.toUnsignedString(getPackIdLong());
+    }
+
+    /**
+     * The ID of the pack the sticker is from.
+     *
+     * @return the ID of the pack the sticker is from
+     */
+    public long getPackIdLong()
+    {
+        return packId;
+    }
+
+    /**
+     * The Discord hash-id of the sticker. This represents the actual asset file in the CDN for the sticker.
+     * <br><b>The URL for fetching sticker assets is currently private.
+     *
+     * @return the Discord hash-id of the sticker
+     */
+    @Nonnull
+    public String getAssetHash()
+    {
+        return asset;
+    }
+
+    /**
+     * The Discord hash-id of the preview of the sticker. This represents the actual preview asset file in the CDN for the sticker.
+     * <br><b>The URL for fetching sticker assets is currently private.
+     *
+     * @return the Discord hash-id of the preview image of the sticker or {@code null} if the sticker has no preview image
+     */
+    @Nullable
+    public String getPreviewAssetHash()
```

Comment:
How would I use this? The other asset hash seems to have a URL but this one does not?

Reply:
well, considering at least from my testing no sticker has a preview asset (yet?), i don't know how to comment on that

Reply:
Then I'd say we just get rid of this for now and comment it out. We can bring it back in the future if it becomes useful.